### PR TITLE
Refactoring of storage of tokens / assets

### PR DESCRIPTION
**Pull request by jameswhollister**
## Ticket
* aa1927b714

## Changes
* Refactoring of storage of tokens / assets

### CODE REVIEW

File: src/renderer/services/neo.js [hunk 1]
```diff
@@ -383,187 +384,151 @@ export default {
...
+                  })
+                  .catch((e) => {
+                    if (e.message.indexOf('Expected a hexstring but got') > -1) {
+                      // console.log(`Removing token due to exception: ${nep5.assetId} ${e}`);
+                      assets.removeUserAsset(holding.assetId);
+                    } else if (e.message.indexOf('Invalid results length!') > -1) {
+                      // console.log(`Removing token due to exception: ${nep5.assetId} ${e}`);
+                    } /* else {
                      console.log(`couldn't fetch token: ${nep5.assetId} ${e}`);
                   } // else mep5balance.needsRefresh will stay true, causing retry getting balance next time */
```
Comment (jsolman) [inline src/renderer/services/neo.js]: I believe this comment is now stale.

File: src/renderer/services/neo.js [hunk 2]
```diff
@@ -383,187 +384,151 @@ export default {
...
+                isUserAsset,
+              };
+            };
 
-              const inMemory = _.get(store.state.nep5Balances, nep5.assetId);
 
-              if (restrictToSymbol && nep5.symbol !== restrictToSymbol) {
-                return;
-              }
+            if (!lastVerifiedTokenBalances
```
Comment (jsolman) [inline src/renderer/services/neo.js]: The needsRefresh flag is no longer being looked at. Isn't it still being set?

Comment (jsolman) [reply src/renderer/services/neo.js]: For example:
[dex.js line 862](https://github.com/aphtoken/aph-wallet-gui/pull/132/files#diff-54fcb2cbbfe08304e72ac626427a1d01R862)
[dex.js line 904](https://github.com/aphtoken/aph-wallet-gui/pull/132/files#diff-54fcb2cbbfe08304e72ac626427a1d01R904)

File: src/renderer/components/TokenSale.vue [hunk 3]
```diff
@@ -64,10 +64,10 @@ export default {
 
   computed: {
     tokens() {
-      const list = this.$services.tokens.getAllAsArray().reduce(
+      const list = this.$services.assets.getAllAsArray().reduce(
         (result, { symbol, name, assetId, network, sale }) => {
           if (!sale || network !== this.$services.network.getSelectedNetwork().net
-              || moment(sale.endDate) < moment().utc()) {
+              || moment(sale.endDate) < moment().utc) {
```
[the remark this answers is not in the document]
Comment (jameswhollister) [reply src/renderer/components/TokenSale.vue]: Correct, fixed this ty.

**Comment by jameswhollister** [comment]
Re-incorporated concept to force a refresh on certain assets, using a list of asset ids that need to be refreshed in the vuex state

File: src/renderer/components/assets/AssetTable.vue [hunk 4]
```diff
@@ -27,8 +27,8 @@ export default {
         return name.toLowerCase().indexOf(searchBy) > -1
           || symbol.toLowerCase().indexOf(searchBy) > -1;
       }).map((holding) => {
-        const canRemove = holding.isCustom === true && holding.symbol !== 'APH'
-          && (holding.balance === 0);
+        const canRemove = holding.isNep5 === true && holding.isUserAsset === true
```
Comment (crishellco) [inline src/renderer/components/assets/AssetTable.vue]: I don't see where else `isUserAsset` is set/used?

**Pull request merged**